javascript/sbp/file_io.js

```diff
@@ -0,0 +1,240 @@
+/**
+ * Copyright (C) 2015 Swift Navigation Inc.
+ * Contact: Joshua Gross <josh@swift-nav.com>
+ * This source is subject to the license found in the file 'LICENSE' which must
+ * be distributed together with this source. All other rights reserved.
+ *
+ * THIS CODE AND INFORMATION IS PROVIDED "AS IS" WITHOUT WARRANTY OF ANY KIND,
+ * EITHER EXPRESSED OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE IMPLIED
+ * WARRANTIES OF MERCHANTABILITY AND/OR FITNESS FOR A PARTICULAR PURPOSE.
+ */
+
+// Automatically generated from piksi/yaml/swiftnav/sbp/file_io.yaml with generate.py.
+// Please do not hand edit!
+
+/**
+Package description:
+
+Messages for using device's onboard flash filesystem
+functionality. This allows data to be stored persistently in the
+device's program flash with wear-levelling using a simple filesystem
+interface. The file system interface (CFS) defines an abstract API
+for reading directories and for reading and writing files.
+
+These are in the implementation-defined range (0x0000-0x00FF), and
+intended for internal-use only. Note that some of these messages
+share the same message type ID for both the host request and the
+device response.
+
+**/
+
+var SBP = require('./sbp');
+var Parser = require('binary-parser').Parser;
+
+/**
+ * SBP class for message MSG_FILEIO_READ_REQ (0x00A8).
+ *
+ * The file read message reads a certain length (up to 255 bytes)from a given offset into a file, and returns the data in aMSG_FILEIO_READ_RESP message where the message length fieldindicates how many bytes were succesfully read.The sequencenumber in the request will be returned in the response.If the message is invalid, a followup MSG_PRINT message willprint "Invalid fileio read message". A device will only respondto this message when it is received from sender ID 0x42.
```

Comment:
resolved - I fixed all the spacing errors (AFAICT). I improved the word wrapping significantly.